### ISSUE TITLE
Adding publication and history dates to articles

### ISSUE THREAD

**Issue by JGilbert-eLife** [substance/texture#535]
This is vital to using Texture as a production tool. We need to be able to add and change a published date

```            <pub-date publication-format="electronic" date-type="pub">
                <day>25</day>
                <month>04</month>
                <year>2016</year>
</pub-date>
           <pub-date pub-type="collection">
                <year>2016</year>
            </pub-date>
<volume>5</volume>
```
as well as including and being able to view dates detailing the article history. These would not need to be editable, but they will have to be visible.

```
            <history>
                <date date-type="received" iso-8601-date="2016-03-01">
                    <day>01</day>
                    <month>03</month>
                    <year>2016</year>
                </date>
                <date date-type="accepted" iso-8601-date="2016-04-01">
                    <day>01</day>
                    <month>04</month>
                    <year>2016</year>
                </date>
</history>
```

**Comment by michael** [comment]
`pub-date pub-type="collection"` is redundant to the primary pub-date. Can we leave it out?

**Comment by Melissa37** [comment]
Yes it is. We have some internal system using it, but it' a hangover.
@gnott FYI future :-)

**Comment by gnott** [comment]
Thanks, eLife relies on this for some logic at the moment. Internally we should look for an alternative again.

**Comment by gustavofonseca** [comment]
`pub-date pub-type="collection"` is not always redundant. There are journals in SciELO that are advancing the publication of issues of the next year, so we need a way to represent the publication data (the date related to the issue) and the online date in SciELO.

**Comment by gustavofonseca** [comment]
ops, sorry for closing.

**Comment by fabiobatalha** [comment]
I agree with @gustavofonseca 
It is one of the Érudit requirements, to handle the issue date that represents the original date of publication of a "printed version", and the publication date in the electronic source which may differs from the original date of publication.

At Érudit we are handling it including a new element pub-date with the attribute @date-type="issue". 

``` xml
      <pub-date date-type="issue">
        <season>automne</season>
        <year>2014</year>
      </pub-date>
      <pub-date publication-format="epub" date-type="pub">
        <day>26</day>
        <month>06</month>
        <year>2014</year>
      </pub-date>
```
We can change it if a common approach is defined to handle this information.

**Comment by Melissa37** [comment]
The versioning JATS4R sub group will reconvene once JATS V2 is officially released - expected in Jan '19. The addition of ```<event>``` to the DTD will mean a recommendation should hopefully be made relatively quickly after that.

Can you need for this wait until then?

M

**Comment by lpanebr** [comment]
PMC tagging guidelines also (currently) require `pub-date pub-type="collection"`.

**Comment by michael** [comment]
We are happy to address this in 2.0 but we need a final specification of all fields we should expose in the interface.

Here's what we have now:

![image](https://user-images.githubusercontent.com/2931/48945510-f3a2d580-ef2a-11e8-8180-07638cc1bf05.png)

Can you provide me what fields are missing (including the JATS tagging and how to label them). Thank you!

**Comment by Melissa37** [comment]
>PMC tagging guidelines also (currently) require pub-date pub-type="collection".

Thanks, @lpanebr, I often forget PMC guidelines! In that case eLife will need to retain it too!

**Comment by Melissa37** [comment]
>Can you provide me what fields are missing (including the JATS tagging and how to label them). 

I can see nothing missing from that list, we at eLife do not publish all those fields but might do in future and they are necessary for many.

Once JATS 2 is out of draft we will also be adding:

```
<pub-history>
                <event>
                    <event-desc>This article was ogininally published as a <ext-link
                            ext-link-type="uri"
                            xlink:href="https://www.biorxiv.org/content/early/2017/03/24/118356"
                            >preprint on Biorxiv</ext-link></event-desc>
                    <date iso-8601-date="2016-03-01">
                        <day>24</day>
                        <month>03</month>
                        <year>2017</year>
                    </date>
                </event>
            </pub-history>
```

We will also start publishing links to previous versions of the article:

```
<pub-history>
                <event>
                    <event-desc>This article was ogininally published as a <ext-link
                        ext-link-type="uri"
                        xlink:href="https://www.biorxiv.org/content/early/2017/03/24/118356"
                        >preprint on Biorxiv</ext-link></event-desc>
                    <date iso-8601-date="2016-03-01">
                        <day>24</day>
                        <month>03</month>
                        <year>2017</year>
                    </date>
                </event>
                <event>
                    <event-desc><ext-link ext-link-type="uri" xlink:href="https://elifesciences.org/articles/35886v1">Accepted Manuscript published</ext-link></event-desc>
                    <date iso-8601-date="2017-07-18">
                        <day>17</day>
                        <month>07</month>
                        <year>2018</year>
                    </date>
                </event>
<event>
                    <event-desc><ext-link ext-link-type="uri" xlink:href="https://elifesciences.org/articles/35886v2">Version of Record published</ext-link></event-desc>
                    <date iso-8601-date="2018-07-30">
                        <day>30</day>
                        <month>07</month>
                        <year>2018</year>
                    </date>
                </event>
            </pub-history>
```

**Comment by michael** [comment]
@Melissa37 I'm still not clear on what we should change in the Texture 2.0 iteration. This `<pub-history>` thing should be a separate discussion, and not critical atm, right?

Assuming that eLife is happy with the current setup, @substance/core-contributors last chance to get your requirements in. :)

**Comment by michael** [comment]
Sorry ping @substance/texture-contributors.

**Comment by fabiobatalha** [comment]
the pub-date can appear in 3 different publication-format's.

``` xml
      <pub-date id="date-5" publication-format="print" date-type="pub" iso-8601-date="1999-01-29">
        <day id="day-5">29</day>
        <month id="month-5">01</month>
        <year id="year-5">1999</year>
      </pub-date>
      <pub-date id="date-5" publication-format="electronic" date-type="pub" iso-8601-date="1999-01-29">
        <day id="day-5">29</day>
        <month id="month-5">01</month>
        <year id="year-5">1999</year>
      </pub-date>
      <pub-date id="date-5" publication-format="collection" date-type="pub" iso-8601-date="1999-01-29">
        <day id="day-5">29</day>
        <month id="month-5">01</month>
        <year id="year-5">1999</year>
      </pub-date>
```

**Comment by lpanebr** [comment]
@fabiobatalha I'm not sure if Erudit is different but PMC guidelines uses "collection" in the **date-type** attribute.[1]

```xml
<pub-date date-type="pub" publication-format="electronic">
  <day>1</day>
  <month>1</month
  <year>2013</year>
</pub-date>
<pub-date date-type="collection" publication-format="electronic">
  <month>1</month>
  <year>2013</year>
</pub-date>
```
[1] https://www.ncbi.nlm.nih.gov/pmc/pmcdoc/tagging-guidelines/article/tags.html#el-pubdate

**Comment by lpanebr** [comment]
I've just realized that in a previous comment **I wrongly stated** that PMC uses `pub-type="collection"`

Sorry... the comment above is the right one.

**Comment by michael** [comment]
What does `date-type=collection` mean vs `date-type=pub`? 

Is it correct that all dates have have format YYYY-MM-DD ?

**Comment by lpanebr** [comment]
`pub` stands for publication of each single article, while `collection` is used to identify the logical group that holds a group of articles. If you take eLife for instance they do not use the Issue but collect articles by month on their website.

PMC's explanation:

> **Articles Published Online and Collected into "Issues" Online Model** — Issues here do not need to be named "Issue 1", "Issue 2", etc. These are just collections of articles online. The collecting might be done by dates (Months, Quarters, Years) or by Volume. For this model, each article will have `<pub-date>` with `@date-type="pub"` and `@publication-format="electronic"` to represent its individual publication date and a `<pub-date>` with `@date-type="collection"` and `@publication-format="electronic"` to represent the date of the online collection it belongs in. This date may be a day/month/year, month/year, season/year, or just year.
https://www.ncbi.nlm.nih.gov/pmc/pmcdoc/tagging-guidelines/article/tags.html#el-pubdate

Yes, the `@iso-8601-date` follows that format: https://jats.nlm.nih.gov/publishing/tag-library/1.1/attribute/iso-8601-date.html

**Comment by michael** [comment]
I tried my best to formalise what has been discussed in this thread. We will send the requirements document by end of week for review (and you can make comments there).

I'm assuming all dates we collect can be represented as an iso-8601-date. This does not include things like `2018/Spring` or `2018/02` etc.
